src/sshagent/OpenSSHKey.cpp

```diff
@@ -336,20 +367,39 @@ bool OpenSSHKey::openPrivateKey(const QString& passphrase)
         }
     }
 
-    BinaryStream keyStream(&rawPrivateData);
+    if (m_privateType == "DSA PRIVATE KEY") {
+        if (!ASN1Key::parseDSA(rawPrivateData, *this)) {
+            m_error = tr("Reading DSA private key failed, only unencrypted keys are supported at this time");
+            return false;
+        }
+
+        return true;
+    } else if (m_privateType == "RSA PRIVATE KEY") {
+        if (!ASN1Key::parseRSA(rawPrivateData, *this)) {
+            m_error = tr("Reading RSA private key failed, only unencrypted keys are supported at this time");
+            return false;
+        }
 
-    quint32 checkInt1;
-    quint32 checkInt2;
+        return true;
+    } else if (m_privateType == "OPENSSH PRIVATE KEY") {
```

Comment:
These strings should be defined as static constexpr somewhere. You use them too often as literals.